bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java

```diff
@@ -0,0 +1,234 @@
+import io.grpc.StatusRuntimeException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.filter.BinaryComparator;
+import org.apache.hadoop.hbase.filter.CompareFilter.CompareOp;
+import org.apache.hadoop.hbase.filter.SingleColumnValueFilter;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.Iterators;
+
+public class TestSingleColumnValueFilter extends AbstractTest {
+
+  byte[] qual = dataHelper.randomData("testSingleValueFilterCompOps");
+  static final int count = 10;
+  static final byte[] COLUMN_FAMILY = Bytes.toBytes("cd");
+  static final byte[] QUALIFIER = Bytes.toBytes("value");
+
+  TableName tableName = IntegrationTests.newTestTableName();
+  private Map<String, Long> result = new HashMap<>();
+  private Table table;
+
+  @Before
+  public void fillTable() throws IOException {
+    try (Admin admin = getConnection().getAdmin()) {
+      if (!admin.tableExists(tableName)) {
+        final HTableDescriptor htableDescriptor = new HTableDescriptor(tableName);
+        HColumnDescriptor hcolDesc = new HColumnDescriptor(COLUMN_FAMILY);
+        htableDescriptor.addFamily(hcolDesc);
+        try {
+          admin.createTable(htableDescriptor);
+        } catch (final StatusRuntimeException e) {
+          if (e.getCause() == null || !e.getCause().getMessage().startsWith("ALREADY_EXISTS")) {
+            throw e;
+          }
+        }
+      }
+    }
+    result.clear();
+    table = getConnection().getTable(tableName);
+    for (long i = 0; i < count; i++) {
+      final UUID rowKey = UUID.randomUUID();
+      byte[] row = Bytes.toBytes(rowKey.toString());
+      Put put = new Put(row);
+      put.addColumn(COLUMN_FAMILY, QUALIFIER, Bytes.toBytes(i));
+      result.put(rowKey.toString(), i);
+      table.put(put);
+    }
+  }
+
+  @After
+  public void delete() throws IOException {
+    try (Admin admin = getConnection().getAdmin()) {
+      if (admin.tableExists(tableName)) {
+        admin.deleteTable(tableName);
+      }
+    }
+    table.close();
+    table = null;
+    result.clear();
+  }
+
+  @Test
+  public void testSingleValueFilterAscii() throws IOException {
+    // Add {, }, and @ to make sure that they do not need to be encoded
+    byte[] rowKeyA = dataHelper.randomData("KeyA{");
```

Comment:
I think it is better to have setup data in @Before, then have test cases:
1. testEqual
2. testGreater
3. testLess
...